### PR TITLE
Add types.is_set() to test whether an arbitrary object is a set as defined by new_sets.bzl.

### PR DESCRIPTION
**Pull request by techsy730**
Since using sets requires special API, it can be useful to determine
whether an object is a set so special handling can be used.
For example, if a method wants to be able to take a list or a set.

### CODE REVIEW

**Comment by techsy730** [comment]
One thing I don't know is whether "new_sets.bzl" is the right place for this method or "types.bzl" is. "types.bzl" seems to be about testing for Bazel's native types, not custom "types".

**Comment by aiuto** [comment]
This looks like it should be in types.bzl not new_set.bzl

File: lib/new_sets.bzl [hunk 1]
```diff
@@ -217,6 +218,19 @@ def _repr(s):
     """
     return repr(s._values.keys())
 
+_SET_TYPE = type(_make())
```
Comment (aiuto) [inline lib/new_sets.bzl]: This should be in types.bzl, alongside the other is_? predicates.

**Comment by techsy730** [comment]
> This looks like it should be in types.bzl not new_set.bzl

Done

File: lib/new_sets.bzl (hunk 1 above)
Comment (techsy730) [reply lib/new_sets.bzl]: For some reason GitHub is stating a change is requested

Comment (techsy730) [reply lib/new_sets.bzl]: Done

**Comment by aiuto** [comment]
This has sat too long. If I do not hear from anyone else I will merge this on 2019-09-06.

**Comment by jin** [comment]
Is this mergeable?

**Pull request merged**